Type/NewPassword.pm

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2023 bivio Software, Inc.  All rights reserved.
+package Bivio::Type::NewPassword;
+use strict;
+use Bivio::Base 'Type.ConfirmPassword';
+
+my($_F) = b_use('IO.File');
+my($_TE) = b_use('Bivio.TypeError');
+
+my($_WEAK_PASSWORDS) = {};
+my($_C) = b_use('IO.Config');
+$_C->register(my $_CFG = {
+    weak_regex => undef,
+    weak_corpus_path => undef,
+    in_weak_corpus => sub {
```

Comment:
I think this should just be weak_corpus, which can be:

- undef: empty list
- scalar: path
- ARRAY: list of passwords

The parser would operate on the lines or `$_WEAK_PASSWORDS`: strip specials, downcase, and make HASH.

The upgrade (later) would be to allow the path to end in bdb, and then open that tied to _WEAK_PASSWORDS so the code wouldn't have to change very much to support a bdb file.

Reply:
I'm not sure there's much utility in the config supporting an array of passwords; I don't think you'd want to specify any reasonable list of weak passwords that way for production. Configuring tests would be slightly easier, but it's trivial to create a temp file with a list of passwords and would provide testing for functionality that might actually be used in production.

Reply:
That's fine. I just saw the regular expression for testing.

Reply:
Well, I think the regex has utility in production, though.

Reply:
I don't think we want to maintain our own corpus. We might add to it, e.g. not allowing bivio.com as a password. I think we should just trust the standard corpi.

Reply:
Yes, but some sites are going to have easily-guessable terms that don't contain the entirety of the domain name. E.g., for Einstein Brothers Bagels, their uri is `einsteinbros.com`, but you might want to exclude `/(einstein|bros|brother|bagel)/`.